### PR TITLE
Fargate support (3rd iteration)

### PR DESCRIPTION
**Pull request by michaelwittig**
Support for AWS Fargate

## Todos
- [x] Cluster
- [x] ALB integration
- [x] ~NLB integration~
- [x] Service with single container/task
- [ ] CW Alarms on HealthyHostCount https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html
- [x] Service with two containers/tasks
- [x] Service with three containers/tasks
- [x] Tests
- [x] Docs

## Changes
* [New Template] `fargate/cluster` - Fargate cluster
* [New Template] `fargate/service-dedicated-alb` -  Fargate service that runs on a Fargate cluster based on fargate/cluster.yaml and uses a dedicated ALB
* [Improvement] `ecs/cluster` - Optional support for ALB access logging and idle timeout
* [Improvement] `ecs/service-dedicated-alb` - Optional support for ALB access logging and idle timeout
* [Improvement] `ecs/service-dedicated-alb` - Added parameters to control `Cpu`, `Memory`, `AutoScaling`, and `HealthCheckGracePeriod`
* [Improvement] `ecs/service-cluster-alb` - Added parameters to control `Cpu`, `Memory`, `AutoScaling`, and `HealthCheckGracePeriod`

### CODE REVIEW

File: fargate/service-dedicated-alb.yaml [hunk 1]
```diff
@@ -0,0 +1,768 @@
...
+  LoadBalancerIdleTimeout:
+    Description: 'The idle timeout value, in seconds.'
+    Type: Number
+    Default: 60
+    MinValue: 1
+    MaxValue: 4000
+  LoadBalancerDeregistrationDelay:
+    Description: 'The amount time (in seconds) to wait before changing the state of a deregistering target from draining to unused.'
+    Type: Number
+    Default: 300
```
Comment (andreaswittig) [inline fargate/service-dedicated-alb.yaml]: I would argue, that 60 seconds should be enough for most of the workloads? Speeds up the deployment process a lot.

File: fargate/service-dedicated-alb.yaml [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+        ToPort: 65535
+        IpProtocol: tcp
+  Service:
+    Type: 'AWS::ECS::Service'
+    DependsOn: HttpListener
+    Properties:
+      Cluster: {'Fn::ImportValue': !Sub '${ParentClusterStack}-Cluster'}
+      DeploymentConfiguration:
+        MaximumPercent: 200
+        MinimumHealthyPercent: 50
```
Comment (andreaswittig) [inline fargate/service-dedicated-alb.yaml]: Why downgrading the capacity during a deployment? Especially, as we are using Fargate? I'd recommend 100%.

Comment (michaelwittig) [reply fargate/service-dedicated-alb.yaml]: good catch

File: fargate/service-dedicated-alb.yaml (hunk 1 above)
Comment (michaelwittig) [reply fargate/service-dedicated-alb.yaml]: I agree

**Pull request merged**